File: UIKitExtensions/UITextField+DDExtensions.m
//
//  UITextField+DDExtensions.m
//  UIKitExtensions
//
//  Copyright © 2019 dashdevs.com. All rights reserved.
//

#import "UITextField+DDExtensions.h"

@implementation UITextField (DDCursor)

- (void)dd_moveCursorToLocation:(NSInteger)location {
    UITextPosition *position = [self positionFromPosition:self.beginningOfDocument offset:location];
    self.selectedTextRange = position ? [self textRangeFromPosition:position toPosition:position] : nil;
}

@end

File: Sources/UITextFieldExtension/UITextField+DDExtensions.swift
//
//  UITextField+DDExtensions.swift
//  
//
//  Created by Anatolij Klimenko on 02.12.2021.
//

import UIKit

extension UITextField {
    
    public func moveCursorToLocation(location: Int) {
        guard let position = position(from: beginningOfDocument, offset: location) else {
            selectedTextRange = nil
            return
        }
        selectedTextRange = textRange(from: position, to: position)
    }
}
